### PR TITLE
Use custom warnings and exceptions

### PR DESCRIPTION
**Pull request by Bjwebb**
https://github.com/OpenDataServices/flatten-tool/issues/450 https://github.com/OpenDataServices/flatten-tool/issues/451

### CODE REVIEW

**Comment by Bjwebb** [comment]
@jpmckinney Does this PR resolve your 2 most recent issues?

**Comment by jpmckinney** [comment]
Looks good, thank you!

File: flattentool/__init__.py [hunk 1]
```diff
@@ -239,11 +240,13 @@ def unflatten(
     """
 
     if input_format is None:
-        raise Exception("You must specify an input format (may autodetect in future")
+        raise FlattenToolError(
+            "You must specify an input format (may autodetect in future"
```
Comment (R2ZER0) [inline flattentool/__init__.py:244]: I'm guessing there's meant to be a ) at the end of the message

**Review by R2ZER0 — APPROVED** [review]
Seems sensible, looks good to me :+1:

**Pull request merged**